grpc/src/test/java/com/linecorp/armeria/client/grpc/GrpcServiceImplErrorFromClientViewTest.java

```diff
@@ -0,0 +1,325 @@
+                                              TestServiceBlockingStub.class)
+                                   .withDeadlineAfter(2, TimeUnit.SECONDS);
+
+                client.unaryCall(REQUEST_MESSAGE);
+            }).satisfies(cause -> {
+                assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.UNKNOWN.getCode());
+                assertThat(Status.trailersFromThrowable(cause)).isNotEqualTo(corruptedMetadata);
+            });
+
+            final RequestLog log = captor.get().log().whenComplete().join();
+            assertThat(log.responseStatus()).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
+            assertThat(log.responseCause())
+                    .satisfies(cause -> {
+                        assertThat(cause).isInstanceOf(StatusRuntimeException.class);
+                        assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(
+                                Status.UNKNOWN.getCode());
+                        assertThat(Status.trailersFromThrowable(cause)).satisfies(metadata -> {
+                            assertThat(isSameSizeAndNameAndValues(metadata, corruptedMetadata)).isFalse();
+                        });
+                    });
+        }
+    }
+
+    // Error inside #onError at server streaming RPC
+    @Test
+    void clientUnaryCallServerStreamingOutputCall() throws InterruptedException {
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            assertThatThrownBy(() -> {
+                final TestServiceBlockingStub client = GrpcClients.newClient(
+                                                                          "http://127.0.0.1:" + serverWithBlockingExecutor.httpPort(),
+                                                                          TestServiceBlockingStub.class)
+                                                                  .withDeadlineAfter(60, TimeUnit.SECONDS);
+                final Iterator<StreamingOutputCallResponse> it = client.streamingOutputCall(
+                        STREAMING_OUTPUT_CALL_REQUEST_MESSAGE);
+                while (it.hasNext()) {
+                    it.next();
+                }
+            }).satisfies(cause -> {
+                assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(Status.CANCELLED.getCode());
+                assertThat(Status.trailersFromThrowable(cause)).isNotEqualTo(corruptedMetadata);
+            });
+
+            final RequestLog log = captor.get().log().whenComplete().join();
+            assertThat(log.responseStatus()).isEqualTo(HttpStatus.OK);
+            assertThat(log.responseCause())
+                    .isInstanceOf(StatusRuntimeException.class)
+                    .satisfies(cause -> {
+                        assertThat(Status.fromThrowable(cause).getCode()).isEqualTo(
+                                Status.CANCELLED.getCode());
```

Comment:
My understanding is that;

Shouldn't we expect `Status.UNKNOWN` or `INTERNAL` , because the main cause is at server side for example?